Sources/PackageManagerDocs/Documentation.docc/CreatingSwiftPackage.md

```diff
@@ -81,3 +81,78 @@ This creates a package with:
 The sample macro, `StringifyMacro`, is documented in the Swift Evolution proposal for [Expression Macros](https://github.com/swiftlang/swift-evolution/blob/main/proposals/0382-expression-macros.md)
 and the WWDC [Write Swift macros](https://developer.apple.com/videos/play/wwdc2023/10166) video.
 For further documentation, see macros in [The Swift Programming Language](https://docs.swift.org/swift-book/documentation/the-swift-programming-language/macros/) book.
+
+### Creating a Package based on a custom template
+
+Swift Package Manager can create packages based on custom templates taht authors distribute as Swift packages.
+These templates can be obtained from local directories, Git repositories, or package registries, and provide interactive configuration through command-line arguments.
+To create a package from a custom template, use the `swift package init` command with the `--type` option along with a template source:
+
+```bash
+# From a package registry
+$ swift package init --type MyTemplate --package-id author.template-example
+
+# From a Git repository
+$ swift package init --type MyTemplate --url https://github.com/author/template-example
+
+# From a local directory
+$ swift package init --type MyTemplate --path /path/to/template
+```
+
+The template prompts you for configuration options during initialization:
```

Comment:
```suggestion
If there are options, the template prompts you for configuration during initialization:
```

(is this correct? I assumed that it's possible to create a template without options)